runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -133,22 +137,36 @@ public SparkPipelineResult run(Pipeline pipeline) {
             finishState,
             isOneOf(PipelineResult.State.STOPPED, PipelineResult.State.DONE));
 
-        // validate assertion succeeded (at least once).
-        int successAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
+        MetricQueryResults queryResults = result.metrics().queryMetrics(
+            MetricsFilter.builder().addNameFilter(
+                MetricNameFilter.named(PAssert.class, PAssert.SUCCESS_COUNTER)).build());
+        long successAssertions = 0;
+        for (MetricResult<Long> res : queryResults.counters()) {
+          successAssertions += res.attempted().longValue();
```

Comment:
The Spark runner does not support committed metrics. @aviemzur what do you think? We could merge this as is, and file a bug to change it once committed metrics are supported.